src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -779,6 +806,47 @@ private static Details getDetailsForOthers(DetailAST ast) {
             return new Details(lcurly, rcurly, getNextToken(ast), true);
         }
 
+        /**
+         * Collects validation details for anonymomus inner class.
+         *
+         * @param ast a {@code DetailAST} value
+         * @return an object containing all details to make a validation
+         */
+        private static Details getDetailsForAnonymousInnerClass(DetailAST ast) {
+
+            DetailAST rcurly = null;
+            DetailAST lcurly = null;
+            DetailAST nextToken;
+
+            DetailAST ansNode = checkIfDeclarationHasAnonymousInnerClass(ast);
+
+            if(ansNode != null){
+                rcurly = ansNode.getLastChild();
+                lcurly = ansNode.getFirstChild();
+                nextToken = getNextToken(ansNode);
+            }
+            else {
+                nextToken = getNextToken(ast.getLastChild());
+            }
+            return new Details(lcurly, rcurly, nextToken, false);
+        }
+
+        private static DetailAST checkIfDeclarationHasAnonymousInnerClass(DetailAST ast){
+        DetailAST ansNode = null;
+
+        DetailAST childNode = ast.getLastChild().getPreviousSibling();
+
+        while(childNode != null){
+            if(childNode.getType() == TokenTypes.OBJBLOCK){
+                ansNode = childNode;
+                break;
+            }
+            childNode = childNode.getLastChild();
+        }
+
+        return ansNode;
+        }
+
```

Comment:
@nrmancuso @romani 

These both methods that I have made are so far working good for detecting anonymous inner classes. There seems to be an issue here at the moment > . I am debugging and will probably fix it on my own.